src/parseable/streams.rs

```diff
@@ -58,19 +59,36 @@ use crate::{
 
 use super::{
     staging::{
-        reader::{MergedRecordReader, MergedReverseRecordReader},
-        writer::Writer,
+        reader::MergedRecordReader,
+        writer::{DiskWriter, Writer},
         StagingError,
     },
-    LogStream,
+    LogStream, ARROW_FILE_EXTENSION,
 };
 
+// NOTE: this requires that custom partition values should not have special characters in their name/value
+static ARROWS_NAME_STRUCTURE: Lazy<Regex> = Lazy::new(|| {
+    Regex::new(r"^[a-zA-Z0-9&=]+\.(?P<front>\S+)\.\d+\.data\.arrows$").expect("Validated regex")
```

Comment:
This requires that the custom partitioning should adhere to not have name/values that aren't alphanumeric. This point has to be documented well.